plugin.go

```diff
@@ -0,0 +1,391 @@
+	Linux           PluginLinux     `json:"Linux,omitempty" yaml:"Linux,omitempty" toml:"Linux,omitempty"`
+	PropagatedMount string          `json:"PropagatedMount,omitempty" yaml:"PropagatedMount,omitempty" toml:"PropagatedMount,omitempty"`
+	Mounts          []Mount         `json:"Mounts,omitempty" yaml:"Mounts,omitempty" toml:"Mounts,omitempty"`
+	Env             []PluginEnv     `json:"Env,omitempty" yaml:"Env,omitempty" toml:"Env,omitempty"`
+	Args            PluginArgs      `json:"Args,omitempty" yaml:"Args,omitempty" toml:"Args,omitempty"`
+}
+
+// PluginDetail .This is a TBD Comments.
+//
+// See https://goo.gl/kaOHGw for more details.
+type PluginDetail struct {
+	ID       string        `json:"Id,omitempty" yaml:"Id,omitempty" toml:"Id,omitempty"`
+	Name     string        `json:"Name,omitempty" yaml:"Name,omitempty" toml:"Name,omitempty"`
+	Tag      string        `json:"Tag,omitempty" yaml:"Tag,omitempty" toml:"Tag,omitempty"`
+	Active   bool          `json:"Active,omitempty" yaml:"Active,omitempty" toml:"Active,omitempty"`
+	Settings PluginSetting `json:"Settings,omitempty" yaml:"Settings,omitempty" toml:"Settings,omitempty"`
+	Config   PluginConfig  `json:"Config,omitempty" yaml:"Config,omitempty" toml:"Config,omitempty"`
+}
+
+// ListPlugins .This is a TBD Comments.
+//
+// See https://goo.gl/kaOHGw for more details.
+func (c *Client) ListPlugins() ([]PluginDetail, error) {
+	resp, err := c.do("GET", "/plugins", doOptions{})
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	pluginDetails := make([]PluginDetail, 0)
+	if err := json.NewDecoder(resp.Body).Decode(&pluginDetails); err != nil {
+		return nil, err
+	}
+	return pluginDetails, nil
+}
+
+// GetPluginPrivileges .This is a TBD Comments.
+//
+// See https://goo.gl/kaOHGw for more details.
+func (c *Client) GetPluginPrivileges(name string) ([]PluginPrivilege, error) {
+	resp, err := c.do("GET", "/plugins/privileges?"+name, doOptions{})
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	var pluginPrivileges []PluginPrivilege
+	if err := json.NewDecoder(resp.Body).Decode(&pluginPrivileges); err != nil {
+		return nil, err
+	}
+	return pluginPrivileges, nil
+}
+
+// InspectPlugins .This is a TBD Comments.
+//
+// See https://goo.gl/kaOHGw for more details.
+func (c *Client) InspectPlugins(name string) (*PluginDetail, error) {
+	resp, err := c.do("GET", "/plugins/"+name+"/json", doOptions{})
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if err != nil {
+		if e, ok := err.(*Error); ok && e.Status == http.StatusNotFound {
+			return nil, &NoSuchPlugin{ID: name}
+		}
+		return nil, err
+	}
+	resp.Body.Close()
+	var pluginDetail PluginDetail
+	if err := json.NewDecoder(resp.Body).Decode(&pluginDetail); err != nil {
+		return nil, err
+	}
+	return &pluginDetail, nil
+}
+
+// RemovePluginOptions .This is a TBD Comments.
+//
+// See https://goo.gl/kaOHGw for more details.
+type RemovePluginOptions struct {
+	// The ID of the container.
```

Comment:
Please fix the field description to be `` The name of the plugin.`` (we can probably get rid of this too, as it's kinda obvious).